src/test/eqrel_datastructure_test.cpp

```diff
@@ -411,7 +411,7 @@ TEST(SparseDjTest, ParallelDense) {
     for (size_t i = 0; i < N; ++i) {
         data_source.push_back(i);
     }
-    std::random_shuffle(data_source.begin(), data_source.end());
+    std::shuffle(data_source.begin(), data_source.end(), std::uniform_int_distribution<size_t>());
```

Comment:
I think you need a generator rather than a distribution, e.g.
```suggestion
    std::random_device rd;
    std::mt19937 generator(rd());

    std::shuffle(data_source.begin(), data_source.end(), generator);
```

Reply:
Correct, but we can get away with just `random_device`, no need for a `mt19937` PRNG.